src/plugins/blocksmith/components/tiktok-embed-block-modal/index.jsx

```diff
@@ -0,0 +1,187 @@
+import DateTimePicker from '@triniti/cms/plugins/blocksmith/components/date-time-picker';
+import Message from '@gdbots/pbj/Message';
+import PropTypes from 'prop-types';
+import React from 'react';
+import TikTokEmbedBlockPreview from '@triniti/cms/plugins/blocksmith/components/tiktok-embed-block-preview';
+import UncontrolledTooltip from '@triniti/cms/plugins/common/components/uncontrolled-tooltip';
+import {
+  Button,
+  Checkbox,
+  FormGroup,
+  Input,
+  Label,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Icon,
+} from '@triniti/admin-ui-plugin/components';
+
+import changedDate from '../../utils/changedDate';
+import changedTime from '../../utils/changedTime';
+import getTikTokId from './getTikTokId';
+
+export default class TikTokEmbedBlockModal extends React.Component {
+  static propTypes = {
+    block: PropTypes.instanceOf(Message).isRequired,
+    isFreshBlock: PropTypes.bool.isRequired,
+    isOpen: PropTypes.bool,
+    toggle: PropTypes.func.isRequired,
+    onAddBlock: PropTypes.func.isRequired,
+    onEditBlock: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+    const { block } = props;
+    this.state = {
+      aside: block.get('aside'),
+      errorMsg: '',
+      hasUpdatedDate: block.has('updated_date'),
+      isValid: block.has('tiktok_id'),
+      touched: false,
+      tiktokId: block.get('tiktok_id'),
+      updatedDate: block.get('updated_date', new Date()),
+    };
+    this.handleAddBlock = this.handleAddBlock.bind(this);
+    this.handleChangeCheckbox = this.handleChangeCheckbox.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTextArea = this.handleChangeTextArea.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+  }
+
+  componentDidMount() {
+    this.inputElement.focus();
+  }
+
+  setBlock() {
+    const { hasUpdatedDate,
+      tiktokId,
+      updatedDate,
+      aside,
+    } = this.state;
+    const { block } = this.props;
+    return block.schema().createMessage()
+      .set('aside', aside)
+      .set('tiktok_id', tiktokId || null)
+      .set('updated_date', hasUpdatedDate ? updatedDate : null);
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeCheckbox({ target: { id, checked } }) {
+    this.setState({ [id]: checked });
+  }
+
+  handleChangeTextArea(event) {
+    let { errorMsg, tiktokId } = this.state;
```

Comment:
@efjacobson @jalopez13 - declare the variables since the destructured value is never used.

Reply:
@arunkarnati updated